command/results_go_test2json.go

```diff
@@ -0,0 +1,364 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+var (
+	testStartRegex = regexp.MustCompile(`^=== RUN\s+.*$`)
+)
+
+const (
+	benchFailPrefix = "--- FAIL: %s"
+	benchSkipPrefix = "--- SKIP: %s"
```

Comment:
surely these constants and the regex above aren't relevant to this format?

Reply:
See point 6 above the struct. test2json does not correctly tag benchmark test names